source/frontend/properties/gc/TevSolver.cpp

```diff
@@ -165,6 +165,59 @@ static bool optimizeNodeBinary(BinExpr& e) {
     return true;
   }
 
+  // Axiom of identity
+  // 0 == 0 -> true
+  // 0 == 1 -> false
+  // 1 == 0 -> false
+  // 1 == 1 -> true
+  if (e.op == ExprOp::Eq) {
+    if ((left_zero && right_zero) || (left_one && right_one)) {
+      to_unary(One);
+      return true;
+    }
+    if ((left_one && right_zero) || (left_zero && right_one)) {
+      to_unary(Zero);
+      return true;
+    }
+  }
```

Comment:
Considering every permutation of X == Y for One, Zero, A, B, C, D, etc is a permutation explosion. A much more sensible approach is to just check if `left_un->val` is equal to `right_un->value`, which encompasses `left_un->val == Zero && right_un->val == Zero`.